File: game-api/src/main/java/org/runestar/client/game/api/GrandExchangeOfferStatus.java
package org.runestar.client.game.api;

public final class GrandExchangeOfferStatus {

    private GrandExchangeOfferStatus() {}

    public static final int STARTING = 1;
    public static final int IDLE = 2;
    public static final int UPDATING = 3;
    public static final int ABORTING = 4;
    public static final int DONE = 5;
}

File: game-api/src/main/java/org/runestar/client/game/api/GrandExchangeOfferStatus.kt
package org.runestar.client.game.api

enum class GrandExchangeOfferStatus(val id: Int) {

    STARTING(1),
    IDLE(2),
    UPDATING(3),
    ABORTING(4),
    DONE(5);

    companion object {

        @JvmField val VALUES = values().asList()

        fun of (id: Int): GrandExchangeOfferStatus {
            return VALUES[id - 1]
        }
    }
}